### PR TITLE
Error handling for websocket connection to janus

### PR DESCRIPTION
**Pull request by fvovan**
@tomaszdurka @vogdb @fvovan 

Cannot resolve host:
```
app error Uncaught exception Error: getaddrinfo ENOTFOUND localhostx
    at errnoException (dns.js:44:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:94:26)
```

Cannot connect:
```
app error Uncaught exception Error: connect ECONNREFUSED
    at exports._errnoException (util.js:746:11)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1010:19)
```

Are we handling connection errors correctly to CM?

### CODE REVIEW

**Comment by fvovan** [comment]
@njam ahh. sorry. Actually in title you meant ws connections to janus but not to CM.
I added very simple handler for ws errors. Just logging and rethrowing. Not sure what to do during these errors. Close browser connection or something else?

**Comment by njam** [comment]
Also note that this made the program exit when we tested it. Would be good if it would continue running!

**Comment by vogdb** [comment]
Start to rework here. Please don't push here

**Comment by vogdb** [comment]
@fvovan please look at `HttpServer.stop` and `JobManager.stop`. Can you try to implement a gentle shutdown in them?

**Comment by vogdb** [comment]
Now working on tests for the new functional.

**Comment by vogdb** [comment]
@fvovan can you review?

File: test/spec/janus-proxy.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+var _ = require('underscore');
+var assert = require('chai').assert;
+require('../helpers/global-error-handler');
+var WebSocket = require('ws');
+var JanusProxy = require('../../lib/janus-proxy');
+var Logger = require('../../lib/logger');
+var Streams = require('../../lib/streams');
+var serviceLocator = require('../../lib/service-locator');
+
+describe.only('JanusProxy', function() {
```
Comment (vogdb) [inline test/spec/janus-proxy.js]: I know, I forgot this `only` again. Will fix

**Comment by vogdb** [comment]
or @tomaszdurka please review?

**Comment by fvovan** [comment]
looks good but probably I would use one array for 
`_connections` and `_connectionListeners`. They are always used together and could be compounded in single array of objects. But most likely is okay now.
Shouldn't we add tests for exception cases?

**Comment by vogdb** [comment]
I agree that one array is better. I couldn't come up with a good name :( Only integration tests can be written for exception cases cause they are attached in `index.js` (only can be attached there)

Also I'm going to remove `stop` functionality from JanusProxy as it overcomplicates things and not necessary if we fix vendor's janus.js library

**Comment by vogdb** [comment]
@fvovan please review

File: lib/http-server.js [hunk 2]
```diff
@@ -97,4 +92,11 @@ HttpServer.prototype.start = function() {
   );
 };
 
+/**
+ * @returns {Promise}
+ */
+HttpServer.prototype.stop = function() {
+  return Promise.resolve();
+};
+
```
Comment (fvovan) [inline lib/http-server.js]: Probably should be deleted  with its call in `index.js` as #84 is closed

File: lib/index.js [hunk 3]
```diff
@@ -55,6 +55,17 @@ Application.prototype.start = function() {
 
   var jobManager = new JobManager(this.config['jobManager']['jobsPath'], []);
   jobManager.start();
+
+  process.on("unhandledRejection", function(reason) {
```
Comment (fvovan) [inline lib/index.js]: do we have conventions to use single quotes for JS code as we have in PHP?

**Comment by fvovan** [comment]
the rest looks good except it has conflicts

**Comment by vogdb** [comment]
@fvovan please look

**Comment by fvovan** [comment]
lgtm

**Pull request merged**